pkg/e2e/e2e.go

```diff
@@ -150,9 +160,29 @@ func beforeSuite() bool {
 		}
 
 		log.Println("Cluster is healthy and ready for testing")
+
+		var kubeconfigBytes []byte
+		if kubeconfigBytes, err = provider.ClusterKubeconfig(viper.GetString(config.Cluster.ID)); err != nil {
+			events.HandleErrorWithEvents(err, events.InstallKubeconfigRetrievalSuccess, events.InstallKubeconfigRetrievalFailure)
+			log.Printf("Failed retrieving kubeconfig: %v", err)
+			getLogs()
+			return false
+		}
+		viper.Set(config.Kubeconfig.Contents, string(kubeconfigBytes))
+
+		if len(viper.GetString(config.Kubeconfig.Contents)) == 0 {
+			// Give the cluster some breathing room.
+			log.Println("OSD cluster installed. Sleeping for 600s.")
+			time.Sleep(600 * time.Second)
+		} else {
+			log.Printf("No kubeconfig contents found, but there should be some by now.")
+		}
```

Comment:
This might be some super dirty legacy code from before we moved to the provider pattern. 

If you look at some job logs, we always hit the `else` condition. It might make more sense just to nuke that whole code block. :thinking:

Reply:
I'm happy for it to be removed.